src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/math/RobustBrentSolverUnitTest.java

```diff
@@ -0,0 +1,60 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.math;
+
+import org.apache.commons.math3.analysis.UnivariateFunction;
+import org.apache.commons.math3.analysis.solvers.BrentSolver;
+import org.apache.commons.math3.exception.NoBracketingException;
+import org.apache.commons.math3.util.FastMath;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+/**
+ * Unit tests for {@link RobustBrentSolver}
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public class RobustBrentSolverUnitTest extends BaseTest {
+
+    private static final double DEF_ABS_ACC = 1e-6;
+    private static final double DEF_REL_ACC = 1e-6;
+    private static final double DEF_F_ACC = 1e-15;
+
+    @Test
+    public void gridTest() {
+        final RobustBrentSolver solver = new RobustBrentSolver(DEF_REL_ACC, DEF_REL_ACC, DEF_F_ACC);
+        final double[] x = solver.createFractalSearchGrid(0, 1, 10, 2);
+        Assert.assertEquals(x.length, 23);
+        Assert.assertEquals(x[0], 0, 1e-12);
+        Assert.assertEquals(x[22], 1, 1e-12);
+    }
+
+    @Test
+    public void detectBracketsTest() {
+        final List<RobustBrentSolver.Bracket> brackets = RobustBrentSolver.detectBrackets(
+                new double[] {0, 1,  2,  3, 4, 5,  6, 7},
+                new double[] {0, 1, -1, -5, 6, 7, -1, 0});
+        Assert.assertEquals(brackets.size(), 5);
+    }
+
+    @Test
+    public void simpleTest() {
```

Comment:
Put in a comment about how this function has multiple zeroes and would not be handled correctly by a regular `BrentSolver`.